### PR TITLE
fix telfhash computation on x64 specific functions

### PR DESCRIPTION
**Pull request by vthib**
The telfhash algorithm excludes x64 functions, see here: https://github.com/trendmicro/telfhash/blob/d2a82f85a5b/telfhash/telfhash.py#L59

Looks like there was a copy paste error when porting this algorithm in YARA, as the test for "is_bad_prefix" is copied, but this is not the right test of "is_x86_64".

This PR properly fixes it.

CC @HoundThe

### CODE REVIEW

File: libyara/modules/elf/elf.c [hunk 1]
```diff
@@ -102,7 +102,7 @@ define_function(telfhash)
         string functions (str.* and mem.*), gcc changes them depending on arch
         symbols starting with . or _ */
     bool is_bad_prefix = name[0] == '.' || name[0] == '_';
-    bool is_x86_64 = name[0] == '.' || name[0] == '_';
+    bool is_x86_64 = strncmp(name + strlen(name) - 2, "64", 2) == 0;
```
Comment (HoundThe) [inline libyara/modules/elf/elf.c:105]: I don't think a symbol name must be non-empty, so `strlen(name)` can return 0 and read into `name - 2` - pre-running the buffer.

**Review by HoundThe — CHANGES_REQUESTED** [review]
Thanks for noticing the copy-paste error and fixing it!

File: libyara/modules/elf/elf.c (hunk 1 above)
Comment (vthib) [reply libyara/modules/elf/elf.c]: I think a non-empty symbol is impossible given we filtered symbols that are global and functions previously, however a one letter symbol is definitely possible, that's a mistake on my part, thanks.

I added a test using custom made elf file with a bit of everything to check the telfhash is correct. I added a one letter symbol and valgrind does indicate out of bound read, as expected.

**Review by HoundThe — APPROVED** [review]
Thanks! LGTM

**Review by plusvic — APPROVED** [review]

**Pull request merged**